server.go

```diff
@@ -159,55 +158,51 @@ func QueryHandler(querier Querier) http.Handler {
 		matchers := make(metric.LabelMatchers, 0, len(req.Matchers))
 		for _, matcher := range req.Matchers {
 			var mtype metric.MatchType
-			switch matcher.GetType() {
-			case generic.MatchType_EQUAL:
+			switch matcher.Type {
+			case MatchType_EQUAL:
 				mtype = metric.Equal
-			case generic.MatchType_NOT_EQUAL:
+			case MatchType_NOT_EQUAL:
 				mtype = metric.NotEqual
-			case generic.MatchType_REGEX_MATCH:
+			case MatchType_REGEX_MATCH:
 				mtype = metric.RegexMatch
-			case generic.MatchType_REGEX_NO_MATCH:
+			case MatchType_REGEX_NO_MATCH:
 				mtype = metric.RegexNoMatch
 			default:
 				http.Error(w, "invalid matcher type", http.StatusBadRequest)
 				return
 			}
-			matcher, err := metric.NewLabelMatcher(mtype, model.LabelName(matcher.GetName()), model.LabelValue(matcher.GetValue()))
+			matcher, err := metric.NewLabelMatcher(mtype, model.LabelName(matcher.Name), model.LabelValue(matcher.Value))
 			if err != nil {
 				http.Error(w, fmt.Sprintf("error creating matcher: %v", err), http.StatusBadRequest)
 				return
 			}
 			matchers = append(matchers, matcher)
 		}
 
-		start := model.Time(req.GetStartTimestampMs())
-		end := model.Time(req.GetEndTimestampMs())
+		start := model.Time(req.StartTimestampMs)
+		end := model.Time(req.EndTimestampMs)
 
 		res, err := querier.Query(ctx, start, end, matchers...)
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
-		resp := &generic.GenericReadResponse{}
+		resp := &ReadResponse{}
 		for _, ss := range res {
-			ts := &generic.TimeSeries{
-				Name: proto.String(string(ss.Metric[model.MetricNameLabel])),
-			}
+			ts := &TimeSeries{}
```

Comment:
The metric name is just another label now in the protobuf (both upstream in the generic write protobufs and in our generic read query ones in Prism).